### PR TITLE
Removed `AgentEnvName.AGENT_VERSION`

### PR DESCRIPTION
**Pull request by nulls**
### What's done:
- removed `AgentEnvName.AGENT_VERSION`
- added `generateVersionFileTaskProvider` to `save-agent`
- used `generated.AGENT_VERSION`

It's extracted from #1916

### CODE REVIEW

**Review by sanyavertolet — APPROVED** [review]

File: gradle/plugins/src/main/kotlin/com/saveourtool/save/buildutils/save-cli-configuration.gradle.kts [hunk 1]
```diff
@@ -51,13 +51,11 @@ dependencies {
     )
 }
 
-// todo: this logic is duplicated between agent and frontend, can be moved to a shared plugin in gradle/plugins
```
Comment (kgevorkyan) [inline gradle/plugins/src/main/kotlin/com/saveourtool/save/buildutils/save-cli-configuration.gradle.kts:54]: now it's not duplicated or what?

File: gradle/plugins/src/main/kotlin/com/saveourtool/save/buildutils/save-cli-configuration.gradle.kts [hunk 2]
```diff
@@ -68,7 +66,6 @@ val generateVersionFileTaskProvider = tasks.register("generateVersionFile") {
             package generated
 
             internal const val SAVE_CORE_VERSION = "$saveCliVersion"
-            internal const val SAVE_CLOUD_VERSION = "$version"
```
Comment (kgevorkyan) [inline gradle/plugins/src/main/kotlin/com/saveourtool/save/buildutils/save-cli-configuration.gradle.kts:71]: was unused?

File: save-agent/build.gradle.kts [hunk 3]
```diff
@@ -30,13 +30,39 @@ kotlin {
         }
     }
 
+    // generate kotlin file with project version
+    val generateVersionFileTaskProvider = tasks.register("generateVersionFile") {
```
Comment (kgevorkyan) [inline save-agent/build.gradle.kts:34]: do already we have this logic in save-cloud somewhere? may be we can create some method, which will expect few arguments like `internal const val AGENT_VERSION = "$version"` etc

**Review by kgevorkyan — APPROVED** [review]

File: save-agent/build.gradle.kts (hunk 3 above)
Comment (kgevorkyan) [reply save-agent/build.gradle.kts:34]: or we have something similar in save-cli?

File: gradle/plugins/src/main/kotlin/com/saveourtool/save/buildutils/save-cli-configuration.gradle.kts (hunk 2 above)
Comment (nulls) [reply gradle/plugins/src/main/kotlin/com/saveourtool/save/buildutils/save-cli-configuration.gradle.kts:71]: yes, we used it in backend\sandbox before to pass only version of save-agent

File: save-agent/build.gradle.kts (hunk 3 above)
Comment (nulls) [reply save-agent/build.gradle.kts:34]: > do already we have this logic in save-cloud somewhere? may be we can create some method, which will expect few arguments like internal const val AGENT_VERSION = "$version" etc

Yes, we have similar logic with frontend

File: gradle/plugins/src/main/kotlin/com/saveourtool/save/buildutils/save-cli-configuration.gradle.kts (hunk 1 above)
Comment (nulls) [reply gradle/plugins/src/main/kotlin/com/saveourtool/save/buildutils/save-cli-configuration.gradle.kts:54]: _frontend_ needs only `SAVE_CLOUD_VERSION` (as `SAVE_VERSION`), _backend_ and _sandbox_ need `SAVE_CORE_VERSION`.
_agent_ needs `AGENT_VERSION` (it's `SAVE_CLOUD_VERSION`).
Now we have similar task between _frontend_ and _agent_ -- will try to combine

**Review by Cheshiriks — APPROVED** [review]

File: save-agent/build.gradle.kts (hunk 3 above)
Comment (nulls) [reply save-agent/build.gradle.kts]: moved to a common place

File: gradle/plugins/src/main/kotlin/com/saveourtool/save/buildutils/save-cli-configuration.gradle.kts (hunk 1 above)
Comment (nulls) [reply gradle/plugins/src/main/kotlin/com/saveourtool/save/buildutils/save-cli-configuration.gradle.kts:54]: done

**Review by 0x6675636b796f75676974687562 — APPROVED** [review]

**Pull request merged**